### ISSUE TITLE
Trying to add a disposable to a DisposableStore that has already been disposed of. The added object will be leaked!

### ISSUE THREAD

**Issue by jrieken** [microsoft/vscode#172620]
Sorry, not steps but I found this in my local log

```
Error: Trying to add a disposable to a DisposableStore that has already been disposed of. The added object will be leaked!
    at DisposableStore.add (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/lifecycle.js:207:34)
    at CustomizedMenuWorkbenchToolBar.setActions (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/platform/actions/browser/toolbar.js:108:42)
    at CustomizedMenuWorkbenchToolBar.updateToolbar (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/editor/contrib/inlineCompletions/browser/inlineSuggestionHintsWidget.js:212:18)
    at CustomizedMenuWorkbenchToolBar.setAdditionalSecondaryActions (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/editor/contrib/inlineCompletions/browser/inlineSuggestionHintsWidget.js:216:18)
    at InlineSuggestionHintsContentWidget.update (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/editor/contrib/inlineCompletions/browser/inlineSuggestionHintsWidget.js:155:26)
    at InlineSuggestionHintsWidget.update (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/editor/contrib/inlineCompletions/browser/inlineSuggestionHintsWidget.js:35:29)
    at vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/editor/contrib/inlineCompletions/browser/inlineSuggestionHintsWidget.js:29:71
    at Listener.invoke (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:523:27)
    at EventDeliveryQueue.deliver (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:697:38)
    at Emitter.fire (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:662:37)
    at vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/editor/browser/widget/codeEditorWidget.js:134:48
    at Listener.invoke (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:523:27)
    at PrivateEventDeliveryQueue.deliver (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:697:38)
    at Emitter.fire (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:662:37)
    at EditorConfiguration._recomputeOptions (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/editor/browser/config/editorConfiguration.js:55:31)
    at EditorConfiguration.updateOptions (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/editor/browser/config/editorConfiguration.js:102:18)
    at CodeEditorWidget.updateOptions (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/editor/browser/widget/codeEditorWidget.js:240:33)
    at TextResourceEditor.updateEditorControlOptions (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/workbench/browser/parts/editor/textCodeEditor.js:33:33)
    at TextResourceEditor.updateEditorConfiguration (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/workbench/browser/parts/editor/textEditor.js:177:22)
    at vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/workbench/browser/parts/editor/textEditor.js:111:72
    at Listener.invoke (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:523:27)
    at EventDeliveryQueue.deliver (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:697:38)
    at Emitter.fire (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:662:37)
    at CodeEditorWidget.setModel (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/editor/browser/widget/codeEditorWidget.js:307:36)
    at vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/editor/browser/widget/codeEditorWidget.js:1144:67
    at Listener.invoke (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:523:27)
    at PrivateEventDeliveryQueue.deliver (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:697:38)
    at Emitter.fire (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:662:37)
    at TextModel.dispose (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/editor/common/model/textModel.js:218:33)
    at WorkbenchModelService.destroyModel (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/editor/common/services/modelService.js:393:29)
    at UntitledTextEditorModel.dispose (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/workbench/common/editor/textEditorModel.js:179:35)
    at vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/workbench/services/textfile/common/textEditorService.js:90:81
    at vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:76:37
    at Listener.invoke (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:523:27)
    at PrivateEventDeliveryQueue.deliver (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:697:38)
    at Emitter.fire (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:662:37)
    at UntitledTextEditorInput.dispose (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/workbench/common/editor/editorInput.js:225:37)
    at UntitledTextEditorInput.dispose (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/workbench/services/untitled/common/untitledTextEditorInput.js:135:19)
    at EditorGroupView.handleOnDidCloseEditor (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/workbench/browser/parts/editor/editorGroupView.js:431:28)
    at EditorGroupView.onDidGroupModelChange (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/workbench/browser/parts/editor/editorGroupView.js:389:30)
    at vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/workbench/browser/parts/editor/editorGroupView.js:368:66
    at Listener.invoke (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:523:27)
    at PrivateEventDeliveryQueue.deliver (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:697:38)
    at Emitter.fire (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/base/common/event.js:662:37)
    at EditorGroupModel.closeEditor (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/workbench/common/editor/editorGroupModel.js:298:40)
    at EditorGroupView.doCloseActiveEditor (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/workbench/browser/parts/editor/editorGroupView.js:1054:36)
    at EditorGroupView.doCloseEditors (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/workbench/browser/parts/editor/editorGroupView.js:1310:22)
    at EditorGroupView.closeEditors (vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/workbench/browser/parts/editor/editorGroupView.js:1271:18)
    at async vscode-file://vscode-app/Users/jrieken/Code/_collab/vscode2/out/vs/workbench/browser/parts/editor/editorCommands.js:676:21
    at async Promise.all (index 0)
add @ lifecycle.ts:270
setActions @ toolbar.ts:169
updateToolbar @ inlineSuggestionHintsWidget.ts:287
setAdditionalSecondaryActions @ inlineSuggestionHintsWidget.ts:292
update @ inlineSuggestionHintsWidget.ts:220
update @ inlineSuggestionHintsWidget.ts:57
(anonymous) @ inlineSuggestionHintsWidget.ts:50
invoke @ event.ts:701
deliver @ event.ts:908
fire @ event.ts:869
(anonymous) @ codeEditorWidget.ts:300
invoke @ event.ts:701
deliver @ event.ts:908
fire @ event.ts:869
_recomputeOptions @ editorConfiguration.ts:101
updateOptions @ editorConfiguration.ts:157
updateOptions @ codeEditorWidget.ts:438
updateEditorControlOptions @ textCodeEditor.ts:50
updateEditorConfiguration @ textEditor.ts:280
(anonymous) @ textEditor.ts:172
invoke @ event.ts:701
deliver @ event.ts:908
fire @ event.ts:869
setModel @ codeEditorWidget.ts:516
(anonymous) @ codeEditorWidget.ts:1606
invoke @ event.ts:701
deliver @ event.ts:908
fire @ event.ts:869
dispose @ textModel.ts:371
destroyModel @ modelService.ts:533
dispose @ textEditorModel.ts:236
(anonymous) @ textEditorService.ts:159
(anonymous) @ event.ts:94
invoke @ event.ts:701
deliver @ event.ts:908
fire @ event.ts:869
dispose @ editorInput.ts:313
dispose @ untitledTextEditorInput.ts:179
handleOnDidCloseEditor @ editorGroupView.ts:603
onDidGroupModelChange @ editorGroupView.ts:553
(anonymous) @ editorGroupView.ts:525
invoke @ event.ts:701
deliver @ event.ts:908
fire @ event.ts:869
closeEditor @ editorGroupModel.ts:485
doCloseActiveEditor @ editorGroupView.ts:1377
doCloseEditors @ editorGroupView.ts:1692
closeEditors @ editorGroupView.ts:1643
await in closeEditors (async)
(anonymous) @ editorCommands.ts:811
closeEditorHandler @ editorCommands.ts:804
handler @ editorCommands.ts:823
invokeFunction @ instantiationService.ts:68
_tryExecuteCommand @ commandService.ts:95
executeCommand @ commandService.ts:60
_doDispatch @ abstractKeybindingService.ts:304
_dispatch @ abstractKeybindingService.ts:192
(anonymous) @ keybindingService.ts:240
Show 18 more frames
```

**Comment by svipas** [comment]
I got this too, and seems like it broke somehow autocomplete, I don't know if it was related or no, but VS Code reload helped. The problem was if I selected some text, pressed <kbd>Tab</kbd> and pressed <kbd>right arrow</kbd> my cursor was back to the text I previously was before autocompleting

- Before completion: `if (state.is|`
- After completion: `if (state.isCampaign|)`
- 🐛 After I press <kbd>right arrow</kbd> it was back to `if (state.is|Campaign` even though it should be moved to the right.

I'm using suggest preview, inline completions with Copilot as well.
